app/scripts/modules/core/src/pipeline/manualExecution/manualPipelineExecution.controller.js

```diff
@@ -109,6 +110,21 @@ module.exports = angular
       this.command.trigger = suppliedTriggerCanBeInvoked ? trigger : _.head(this.triggers);
     };
 
+    const updatePipelinePlan = pipeline => {
+      if (pipeline.type === 'templatedPipeline' && (pipeline.stages === undefined || pipeline.stages.length === 0)) {
+        PipelineTemplateReader.getPipelinePlan(pipeline)
```

Comment:
`getPipelinePlan` takes an optional second parameter `executionId`. Ideally I should've used the build selected in the build drop down if available, but that turned out to be a rabbit hole. The builds in that list only have the CI build number, not the `executionId`. The `executions` variable only have a few executions for each pipeline (depending on the number of pipelines being displayed). And since the dropdown is a react component, I have to use a $watch expression to detect when it change.
So the current implementation will only consider the newest version of the pipeline when deciding wether to show the force rebake box or not. But I think this is Good Enough™, for the same reasons mentioned earlier. It's certainly better than never showing it.